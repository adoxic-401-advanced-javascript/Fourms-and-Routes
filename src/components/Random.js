import React from 'react';
import Characters from './Characters';
import fetchXfile from '../utils/fetch';

export default function Random() {
  const [state, setState] = React.useState([]);

  React.useEffect(() => {
    handleClick();
  }, []);

  const handleClick = () => {
    fetchXfile(`http://xfiles-api.herokuapp.com/api/v1/characters?perPage=1&page=${Math.floor(Math.random() * 462)}`)
      .then(res => {
        setState(res.results);
      });
  };

  return (
    <>
      <Characters title="random" characters={state} />
      <button onClick={handleClick}>Another Random?</button>
    </>
  );
}
