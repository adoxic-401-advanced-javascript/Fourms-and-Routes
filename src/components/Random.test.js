import React from 'react';
import { shallow } from 'enzyme';
import Random from './Random';

describe('Random Test', () => {
  it('Renders an Random', () => {
    const wrapper = shallow(<Random 
      title={'title'}
      characters={['boo']}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
