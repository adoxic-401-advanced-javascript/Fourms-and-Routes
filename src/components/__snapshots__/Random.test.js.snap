// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Test Renders an Random 1`] = `
<Fragment>
  <Characters
    characters={Array []}
    title="random"
  />
  <button
    onClick={[Function]}
  >
    Another Random?
  </button>
</Fragment>
`;
